core/src/main/java/org/apache/accumulo/core/client/impl/Namespace.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.client.impl;
+
+import org.apache.accumulo.core.client.Instance;
+
+public class Namespace {
```

Comment:
@milleruntime took based this implementation on my [ACCUMULO-2589] branch, as I had suggested in the comments of this JIRA issue ([ACCUMULO-3238]). The longer term goal with my branch was to make `Table` and `Namespace` first class citizens in the public API. The ID would be just one facet of those API classes. At this point, it doesn't matter, because they are being used only for internal code. However, I do think it'd be easier for follow-on work to keep the inner-class, in case more public-facing features are adapted from my branch later.

[ACCUMULO-2589]: https://issues.apache.org/jira/browse/ACCUMULO-2589
[ACCUMULO-3238]: https://issues.apache.org/jira/browse/ACCUMULO-3238